cdm/core/src/test/java/thredds/inventory/TestCollectionSpecParser.java

```diff
@@ -58,10 +58,10 @@ public static List<Object[]> getTestParameters() throws URISyntaxException {
         {"/data/ldm/pub/decoded/netcdf/surface/metar/**/", "/data/ldm/pub/decoded/netcdf/surface/metar", true, null,
             null},
 
-        {"/data/ldm/pub/decoded/netcdf/surface/metar/**/*", "/data/ldm/pub/decoded/netcdf/surface/metar", true, null,
+        {"/data/ldm/pub/decoded/netcdf/surface/metar/**/.*", "/data/ldm/pub/decoded/netcdf/surface/metar", true, ".*",
```

Comment:
I very slightly changed the behavior of the original `CollectionSpecParser`. Previously, if the regEx was only one character (here '*' but could be any character) then it resulted in `filter = null`. Now, a regex of '*' will be compiled into a `Pattern` and will fail.

Reply:
hmm, I'm a little worried that will break some users, I think the single '*' pattern is pretty commonly used...

Reply:
It just sort of seemed like a mistake that a one character filter was ignored. You can achieve the same results with no filter or with ".*".

But regardless that could be annoying for users already using a "*" so I can change it back.

Reply:
yeah, I think '**/*' is pretty standard for "anything at any nesting level", even if it is unnecessary

Reply:
back to original behavior in 35143be7db